astropy/coordinates/tests/test_exceptions.py

```diff
@@ -11,25 +11,21 @@
 from astropy import units as u
 from astropy.coordinates import (
     GCRS,
-    ICRS,
-    Galactic,
     NonRotationTransformationError,
     NonRotationTransformationWarning,
+    SkyCoord,
 )
 
 if TYPE_CHECKING:
     from astropy.coordinates import BaseCoordinateFrame
 
 
 class FrameDescription(NamedTuple):
-    frame: BaseCoordinateFrame
+    frame: BaseCoordinateFrame | SkyCoord
```

Comment:
If I understand correctly, what matters is what these classes have in common, so I would suggest to formally define this common base as a `typing.Protocol` and use it here instead of an Union.

Reply:
I actually like that @eerovaher has kept to the simple & readable for now; that seemed to be the conclusion too from the long discussion we had on typing, that we use it first mostly as documentation.

Reply:
that's what I'm proposing too. A protocol would document exactly which parts of these classes are needed here.

Reply:
But it might well be less readable - even a simple `SkyCoordLike` is less clear than just listing two options.

Anyway, it is no big deal, I just want to avoid overzealous typing when the benefits are unclear (and it seems obvious that the whole ecosystem is still rapidly evolving).